Sources/NIOTLS/NIOTypedApplicationProtocolNegotiationHandler.swift

```diff
@@ -74,14 +77,18 @@ public final class NIOTypedApplicationProtocolNegotiationHandler<NegotiationResu
     /// - Parameter alpnCompleteHandler: The closure that will fire when ALPN
     ///   negotiation has completed.
     @_spi(AsyncChannel)
-    public convenience init(eventLoop: EventLoop, alpnCompleteHandler: @escaping (ALPNResult) -> EventLoopFuture<NIOProtocolNegotiationResult<NegotiationResult>>) {
-        self.init(eventLoop: eventLoop) { result, _ in
+    public convenience init(alpnCompleteHandler: @escaping (ALPNResult) -> EventLoopFuture<NIOProtocolNegotiationResult<NegotiationResult>>) {
+        self.init { result, _ in
             alpnCompleteHandler(result)
         }
     }
 
-    deinit {
-        switch self.stateMachine.deinitHandler() {
```

Comment:
Should we drop the `deinitHandler` method?

Reply:
Yes and the method is dropped from the state machine in this PR.